internal/resources/sourcesecret/spec/spec.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package spec
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
+)
+
+var (
+	SpecSchema = &schema.Schema{
+		Type:        schema.TypeList,
+		Description: "",
+		Required:    true,
+		MinItems:    1,
+		MaxItems:    1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				dataKey: dataSchema,
+			},
+		},
+	}
+
+	specsAllowed = [...]string{usernamePasswordKey, sshKey}
```

Comment:
shouldn't it be credentialTypesAllowed?

Similarly, update others.

Reply:
updated